tests/distribute_rewards.rs

```diff
@@ -0,0 +1,195 @@
+#![cfg(feature = "test-sbf")]
+
+mod setup;
+
+use {
+    paladin_rewards_program::{instruction::distribute_rewards, state::HolderRewardsPool},
+    setup::{setup, setup_holder_rewards_pool_account, setup_system_account},
+    solana_program_test::*,
+    solana_sdk::{
+        account::AccountSharedData,
+        instruction::InstructionError,
+        pubkey::Pubkey,
+        signature::Keypair,
+        signer::Signer,
+        transaction::{Transaction, TransactionError},
+    },
+};
+
+#[tokio::test]
+async fn fail_payer_not_signer() {
+    let holder_rewards_pool = Pubkey::new_unique();
+    let payer = Keypair::new();
+    let amount = 500_000_000_000;
+
+    let mut context = setup().start_with_context().await;
+
+    let mut instruction = distribute_rewards(&payer.pubkey(), &holder_rewards_pool, amount);
+    instruction.accounts[0].is_signer = false; // Not signer.
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[instruction],
+        Some(&context.payer.pubkey()),
+        &[&context.payer], // Missing payer.
+        context.last_blockhash,
+    );
+
+    let err = context
+        .banks_client
+        .process_transaction(transaction)
+        .await
+        .unwrap_err()
+        .unwrap();
+
+    assert_eq!(
+        err,
+        TransactionError::InstructionError(0, InstructionError::MissingRequiredSignature)
+    );
+}
+
+#[tokio::test]
+async fn fail_holder_rewards_pool_incorrect_owner() {
+    let holder_rewards_pool = Pubkey::new_unique();
+    let payer = Keypair::new();
+    let amount = 500_000_000_000;
+
+    let mut context = setup().start_with_context().await;
+    setup_system_account(&mut context, &payer.pubkey(), amount).await;
+
+    // Set up a holder rewards pool account with incorrect owner.
+    {
+        context.set_account(
+            &holder_rewards_pool,
+            &AccountSharedData::new_data(
+                100_000_000,
+                &vec![0; 165],
```

Comment:
micro-nit: maybe at least use `8` so it can be properly deserialized as a rewards pool
```suggestion
                &vec![0; 8],
```

Reply:
Sounds good. I only did this because the owner check comes before deserialization.